Bearded.Utilities/Monads/Result.cs

```diff
@@ -1,16 +1,19 @@
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Runtime.CompilerServices;
 
 namespace Bearded.Utilities.Monads
 {
     public readonly struct Result<TResult, TError> : IEquatable<Result<TResult, TError>>
     {
         private readonly bool isSuccess;
+        [AllowNull]
         private readonly TResult result;
+        [AllowNull]
         private readonly TError error;
 
-        private Result(bool isSuccess, TResult result, TError error)
+        private Result(bool isSuccess, [AllowNull] TResult result, [AllowNull] TError error)
```

Comment:
Are these necessary? Same with the annotations on the fields? Are they there so we can pass `default` in the static construction methods without issues?